plugins/flatpak/gs-flatpak.c

```diff
@@ -1388,10 +1469,29 @@ gs_flatpak_add_updates_pending (GsFlatpak *self, GsAppList *list,
 			g_warning ("failed to add flatpak: %s", error_local->message);
 			continue;
 		}
-		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
+
+		if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME)
+			main_app = get_main_app_of_related (self, app, cancellable,
+							    &error_local);
+
+		if (main_app == NULL) {
+			if (error_local != NULL &&
+			    !g_error_matches (error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND)) {
```

Comment:
I think a comment here explaining why you make an exception for `_NOT_FOUND` (i.e. not all runtimes are extensions) would be helpful to understand this line.

Also, if `error_local != NOT_FOUND` is considered a legit problem, I think that should be printed with a `g_warning()`, not `g_debug()` (which you could use it if you want for an `else` branch.